### ISSUE TITLE
Getting "passsword incorrect" after typing in the sign-in authorization code 

### ISSUE THREAD

**Issue by grigoryk** [mozilla/fxa-content-server#5170]
My browser had a wrong FxA password in the password manager. I was able to use it to get to the "typ in authorization code" screen. After entering my code, I get a "password incorrect" email. Typing in a correct password this time, I _still_ need to enter yet another authorization code.

My first impression is that this flow is rather backwards.

**Comment by rfk** [comment]
> My first impression is that this flow is rather backwards.

IOW, were you expecting us to tell you that your password was incorrect before we forced you through this authorization flow?

The "type in authorization code" screen essentially means that we're refusing to even check your password because we think you're an attacker, so you have to prove that you own the account before you can take any more guesses at the password.  But that's pretty hard to message in a user-friendly way :-(

> I still need to enter yet another authorization code

This, however, we could fix if we invested the time to do so - basically allow you to make multiple login attempts authorized by a single code.

**Comment by grigoryk** [comment]
I see. To me this seemed like a routine "2fa" style login, although one I hadn't encountered before, and so behaviour was confusing.

If the messaging is more explicit, e.g. pointing out that password will be checked after correct code is provided, because of "increased security due to suspicious activity", that might be more predictable. But of course, as you say, how do you talk about suspicious activity without being alarmist.

Perhaps if the false-positive rates for these checks are low enough, we _should_ sound alarmist?

**Comment by rfk** [comment]
> To me this seemed like a routine "2fa" style login,

Yeah, I see what you mean - the current flow doesn't really give you any clue that something went wrong, it looks like your password was accepted and you're now at a second step of the procedure.

/cc @ryanfeeley @davismtl since we've talked about the signin-unblock experience a few times recently

**Comment by vladikoff** [comment]
I got another bug report on Slack about this: 

> So I went to log in with my Firefox account. Typed my email + password, then it sent me a 2FA email. I put the code in and it returns that I entered my password incorrectly. Which I actually did make a typo (a couple times, oops), but it tries to 2FA before validating pw.

> Is that normal?

**Comment by shane-tomlinson** [comment]
My hope is that we can use [email first](https://github.com/mozilla/fxa-content-server/pull/5177) to block the user *before* they get to enter their password. If we do, this confusion will evaporate!

**Comment by shane-tomlinson** [comment]
> My hope is that we can use email first to block the user before they get to enter their password. If we do, this confusion will evaporate!

This will require a change to how unblock codes are accepted, but if we can do that, all the unblock confusion with an invalid password will disappear.

**Comment by vladikoff** [comment]
Another report: https://github.com/mozilla/fxa-bugzilla-mirror/issues/299

**Comment by rfk** [comment]
> I never remember my passwords for low priority stuff

ouch 💔

**Comment by davismtl** [comment]
The main pain point I see is that we ask you to confirm your email with every password guess. This is silly. If the user confirmed their email once in the flow, no need to do it again.

And then, to @shane-tomlinson 's point, asking the email first without the password, should allow us to send a confirmation code before they even type in their password. But again, let's NOT ask them to type it in more than once.

**Comment by rfk** [comment]
> This is silly. If the user confirmed their email once in the flow, no need to do it again.

@davismtl is this important enough to spin off a separate bug for it and include in the Q3 quality backlog?  Your comment suggests yes, but let's be explicit about these things given that we want to "lock down" the OKR lists this week.

**Comment by davismtl** [comment]
I filed another bug.
mozilla/fxa-auth-server#1998

**Comment by vladikoff** [comment]
WIP PR in mozilla/fxa-auth-db-mysql#265
